Code/Tools/Libs/GuiFoundation/ContainerWindow/ContainerWindow.cpp

```diff
@@ -10,78 +10,56 @@
 #include <QTabBar>
 #include <QTimer>
 #include <ToolsFoundation/Application/ApplicationServices.h>
+#include <ads/DockManager.h>
+#include <ads/DockWidgetTab.h>
+#include <ads/FloatingDockContainer.h>
 
-ezDynamicArray<ezQtContainerWindow*> ezQtContainerWindow::s_AllContainerWindows;
+ezQtContainerWindow* ezQtContainerWindow::s_pContainerWindow = nullptr;
 bool ezQtContainerWindow::s_bForceClose = false;
 
-ezQtContainerWindow::ezQtContainerWindow(ezInt32 iUniqueIdentifier)
+ezQtContainerWindow::ezQtContainerWindow()
 {
-  m_iUniqueIdentifier = iUniqueIdentifier;
+  setMinimumSize(QSize(800, 600));
   m_bWindowLayoutRestored = false;
   m_pStatusBarLabel = nullptr;
   m_iWindowLayoutRestoreScheduled = 0;
 
-  s_AllContainerWindows.PushBack(this);
+  s_pContainerWindow = this;
 
-  setObjectName(GetUniqueName().GetData());
+  setObjectName("ezEditor");
```

Comment:
Several hardcoded values here. Can we do better?

Reply:
Now that the ezQtContainerWindow is a singleton it is no longer needed to dynamically construct any names and all names become constants. If the constants are only used once I'd argue that moving them elsewhere only reduces readability.
The only thing that remains are the constants used for the ezSettings. As these are only used in the save and restore function I left them alone as it would generate more code to separate them out then is currently present in those functions so I'd argue it is of questionable value. However, I have deduplicated some of the shared code between those functions which cuts down on code size and constants.